cmd/image/import/import.go

```diff
@@ -115,64 +89,107 @@ pvsadm image import -n upstream-core-lon04 -b <BUCKETNAME>  --object rhel-83-100
 			//Find CosInstance of the bucket
-			var svcs []models.ServiceInstanceV2
-			svcs, err = bxCli.ResourceClientV2.ListInstances(controllerv2.ServiceInstanceQuery{
-				Type: "service_instance",
-			})
+			listServiceInstanceOptions := &resourcecontrollerv2.ListResourceInstancesOptions{
+				Type: ptr.To(cloudObjectStorage),
+			}
+			workspaces, _, err := pvsClient.ResouceControllerClient.ListResourceInstances(listServiceInstanceOptions)
 			if err != nil {
-				return err
+				return fmt.Errorf("failed to list the resource instances: %v", err)
 			}
-			cosInstanceName, cosID, crn := findCOSInstanceDetails(svcs, bxCli)
-			if cosInstanceName == "" {
-				return fmt.Errorf("failed to find the COS instance for the bucket mentioned: %s", opt.BucketName)
+
+			getServiceInstanceOptions := &resourcecontrollerv2.GetResourceInstanceOptions{
+				// TODO: possibility of workspaces to either be of type service_instance or composite_instance.
+				ID: workspaces.Resources[0].ID,
 			}
 
-			keys, err := bxCli.GetResourceKeys(cosID)
+			cosInstance, _, err := pvsClient.ResouceControllerClient.GetResourceInstance(getServiceInstanceOptions)
 			if err != nil {
-				return fmt.Errorf("failed to list the service credentials: %v", err)
+				return fmt.Errorf("failed to list the resource instances: %v", err)
 			}
 
-			var cred map[string]interface{}
+			key, _, err := pvsClient.ResouceControllerClient.GetResourceKey(
+				&resourcecontrollerv2.GetResourceKeyOptions{
+					ID: cosInstance.ID,
+				},
+			)
+			if err != nil {
+				return fmt.Errorf("cannot find resource")
+			}
 			var ok bool
-			if len(keys) == 0 {
+
+			// Create the service credential if does not exist
+			if key.ID == nil {
 				if opt.ServiceCredName == "" {
-					opt.ServiceCredName = serviceCredPrefix + "-" + cosInstanceName
+					opt.ServiceCredName = serviceCredPrefix + "-" + *cosInstance.Name
 				}
 
-				// Create the service credential if does not exist
 				klog.V(2).Infof("Auto Generating the COS Service credential for importing the image with name: %s", opt.ServiceCredName)
-				CreateServiceKeyRequest := controller.CreateServiceKeyRequest{
-					Name:       opt.ServiceCredName,
-					SourceCRN:  crn,
-					Parameters: map[string]interface{}{"HMAC": true},
-				}
-				newKey, err := bxCli.ResourceServiceKey.CreateKey(CreateServiceKeyRequest)
+				roleCRN := "crn:v1:bluemix:public:iam::::serviceRole:Writer"
+				params := &resourcecontrollerv2.ResourceKeyPostParameters{}
+				params.SetProperty("HMAC", true)
+
+				key, _, err := pvsClient.ResouceControllerClient.CreateResourceKey(
+					&resourcecontrollerv2.CreateResourceKeyOptions{
+						Name:       ptr.To(opt.ServiceCredName),
+						Source:     cosInstance.CRN,
+						Role:       ptr.To(roleCRN),
+						Parameters: params,
+					},
+				)
 				if err != nil {
-					return err
+					return fmt.Errorf("unable to create resource key for service instance: %s", err.Error())
+				}
+
+				if key.Credentials != nil {
```

Comment:
Change the order to early return
if key.Credentials == nil {
return
}

Reply:
Addressed.